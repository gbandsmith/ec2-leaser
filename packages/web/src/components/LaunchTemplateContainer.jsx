import LaunchTemplate from './LaunchTemplate';

const LaunchTemplateContainer = ({
  data,
  search,
  selectedLaunchTemplateId,
  setLaunchTemplate
}) => {
  return (
    <div className="flex flex-wrap -m-4">
      {data
        ?.filter(
          lt =>
            lt.name.toLowerCase().includes(search.toLowerCase()) ||
            lt === selectedLaunchTemplateId
        )
        .map(({ id, name }) => {
          const selected = selectedLaunchTemplateId === id;
          return (
            <LaunchTemplate
              key={id}
              id={id}
              name={name}
              selected={selected}
              setLaunchTemplate={setLaunchTemplate}
            />
          );
        })}
    </div>
  );
};

export default LaunchTemplateContainer;
