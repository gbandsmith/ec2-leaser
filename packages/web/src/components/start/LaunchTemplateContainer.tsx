import { LaunchTemplateType } from '../../models/LaunchTemplate';
import LaunchTemplate from './LaunchTemplate';

const LaunchTemplateContainer = ({
  data,
  search,
  selectedLaunchTemplateId,
  setLaunchTemplate
}: {
  data: LaunchTemplateType[];
  search: string;
  selectedLaunchTemplateId: string | null;
  setLaunchTemplate: Function;
}) => {
  return (
    <div className="flex flex-wrap -m-4">
      {data
        ?.filter(
          lt =>
            lt.name.toLowerCase().includes(search.toLowerCase()) ||
            lt.id === selectedLaunchTemplateId
        )
        .map(({ id, name }) => {
          const selected = selectedLaunchTemplateId === id;
          return (
            <LaunchTemplate
              key={id}
              id={id}
              name={name}
              selected={selected}
              setLaunchTemplate={setLaunchTemplate}
            />
          );
        })}
    </div>
  );
};

export default LaunchTemplateContainer;
