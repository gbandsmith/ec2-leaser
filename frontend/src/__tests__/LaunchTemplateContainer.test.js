import LaunchTemplateContainer from '../components/LaunchTemplateContainer';
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
jest.mock('react-query', () => ({
  useQuery: () => ({ data: {description: 'mocked desc'}, isLoading: false, error: undefined })
}));

describe('LaunchTemplateContainer', () => {
  let data;
  let search;
  let selectedLaunchTemplateId;
  let setLaunchTemplate;
  let component;

  beforeEach(() => {
      data = [
          { id: 1, name: 'Template A', version: 1 },
          { id: 2, name: 'Template B', version: 2 },
          { id: 3, name: 'Template C', version: 3 },
      ];
      search = '';
      selectedLaunchTemplateId = 1;
      setLaunchTemplate = jest.fn();
      component = render(
          <LaunchTemplateContainer
              data={data}
              search={search}
              selectedLaunchTemplateId={selectedLaunchTemplateId}
              setLaunchTemplate={setLaunchTemplate}
          />
      );
  });

  it('should render all launch templates', () => {
      const { getByText } = component;
      expect(getByText('Template A')).toBeInTheDocument();
      expect(getByText('Template B')).toBeInTheDocument();
      expect(getByText('Template C')).toBeInTheDocument();
  });

  it('should filter launch templates by search', () => {
      const { getByText, queryByText } = component;
      expect(getByText('Template A')).toBeInTheDocument();
      expect(getByText('Template B')).toBeInTheDocument();
      expect(getByText('Template C')).toBeInTheDocument();

      search = 'Template A';
      component.rerender(
          <LaunchTemplateContainer
              data={data}
              search={search}
              selectedLaunchTemplateId={selectedLaunchTemplateId}
              setLaunchTemplate={setLaunchTemplate}
          />
      );

      expect(getByText('Template A')).toBeInTheDocument();
      expect(queryByText('Template B')).not.toBeInTheDocument();
      expect(queryByText('Template C')).not.toBeInTheDocument();
  });

  it('should call setLaunchTemplate when a template is clicked', () => {
      const { getByText } = component;
      fireEvent.click(getByText('Template B'));
      expect(setLaunchTemplate).toHaveBeenCalledWith({"id": 2, "name": "Template B"});
  });

  it('should render with sorted data', () => {
    cleanup();
    const sortedData = [
      { id: 3, name: 'Template C' },
      { id: 2, name: 'Template B' },
      { id: 1, name: 'Template A' }
    ];
    const props = {
      data: sortedData,
      search: '',
      selectedLaunchTemplateId: null,
      setLaunchTemplate: jest.fn(),
    };
    const { getByText } = render(<LaunchTemplateContainer {...props} />);

    expect(getByText('Template C')).toBeInTheDocument();
    expect(getByText('Template B')).toBeInTheDocument();
    expect(getByText('Template A')).toBeInTheDocument();
  });
});
