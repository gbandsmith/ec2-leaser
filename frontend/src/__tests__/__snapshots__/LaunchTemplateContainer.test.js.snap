// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchTemplateContainer should render with sorted data 1`] = `
<div
  className="flex flex-wrap -m-4"
>
  <div
    className="p-2 md:w-1/2 xl:w-1/3"
  >
    <div
      className="p-4 border border-gray-200 rounded-lg cursor-pointer "
      onClick={[Function]}
    >
      <div
        className="h-24"
      >
        <h2
          className="title-font mb-2 text-gray-900 text-lg font-medium"
        >
          John
        </h2>
        <p
          className="text-base leading-relaxed"
        >
          mocked desc
        </p>
      </div>
    </div>
  </div>
  <div
    className="p-2 md:w-1/2 xl:w-1/3"
  >
    <div
      className="p-4 border border-gray-200 rounded-lg cursor-pointer "
      onClick={[Function]}
    >
      <div
        className="h-24"
      >
        <h2
          className="title-font mb-2 text-gray-900 text-lg font-medium"
        >
          Mary
        </h2>
        <p
          className="text-base leading-relaxed"
        >
          mocked desc
        </p>
      </div>
    </div>
  </div>
  <div
    className="p-2 md:w-1/2 xl:w-1/3"
  >
    <div
      className="p-4 border border-gray-200 rounded-lg cursor-pointer bg-yellow-500"
      onClick={[Function]}
    >
      <div
        className="h-24"
      >
        <h2
          className="title-font mb-2 text-gray-900 text-lg font-medium"
        >
          Mike
        </h2>
        <p
          className="text-base leading-relaxed"
        >
          mocked desc
        </p>
      </div>
    </div>
  </div>
</div>
`;
